firefox-ios/Client/metrics.yaml

```diff
@@ -4880,7 +4880,96 @@ places_history_migration:
       - teshaq@mozilla.com
       - sync-team@mozilla.com
     expires: "2024-06-01"
-
+# Address autofill
+addresses:
+  form_detected:
+    type: event
+    description: |
+      Recorded when a form is recognized as an address form.
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/18980
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/19228
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2024-10-01"
+  autofill_prompt_shown:
+    type: event
+    description: |
+      Recorded when autofill popup is shown, indicating which field triggered this event.
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/18980
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/19228
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2024-10-01"
+  autofill_prompt_expanded:
+    type: event
+    description: |
+      Recorded when the autofill prompt was expanded.
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/18980
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/19228
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2024-10-01"
+  autofill_prompt_dismissed:
+    type: event
+    description: |
+      Recorded when the autofill prompt was dismissed.
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/18980
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/19228
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2024-10-01"
+  autofilled:
+    type: event
+    description: |
+      Recorded when a form is autofilled.
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/18980
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/19228
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2024-10-01"
+  modified:
+    type: event
+    description: |
+      Recorded when a field is autofilled and then modified by the user.
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/18980
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/19228
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2024-10-01"
+  settings_autofill:
+    type: event
+    description: |
+      Recorded when the user has tapped Autofill in the settings menu.
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/18980
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/19228
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2024-10-01"
+  saved_all:
+    type: counter
+    description: |
+      A counter of the number of all addresses that are currently saved by the user.
```

Comment:
This is cumulative, though, and we're expecting it to be cumulative, right?

Reply:
I think so. Up until now I noticed counters were used to represent taps or how many times you opened something. This time my understanding is to report how many addresses a user has.

Reply:
But, if we're adding to a counter, and the counter is at, say, 15, and you add, say, count = 3, the new counter is 18, no? And if the user removes two, so count = 1, the new counter will be 19, which is not total they have now? @travis79 am I mistaken here?

Reply:
That's correct, a counter here probably isn't the right metric type. I would recommend a [QuantityMetricType](https://mozilla.github.io/glean/book/reference/metrics/quantity.html) for this. In that case, you would also need to keep in mind the scheduling of the ping, if this goes on the metrics ping (the default transport), then it would only capture the last recorded value as a Quantity type. 

Depending on what you want to do, perhaps a CustomDistribution metric type might be appropriate. Using a distribution would allow you to "sample" the number of addresses many times (once per session?) and all of these measurements would be contained in a single metric that can still be sent on the metrics ping, once a day.

Reply:
Thanks for the sugestions! It seems quantity metric is the way to go, being implemented also for credit cards and logins.